gov2/cross_service/wordfreq/wordfreq/shared/types.go

```diff
@@ -0,0 +1,120 @@
+package shared
+
+import (
+	"log"
+	"math/rand"
+	"time"
+
+	s3events "example.aws/wordfreq/shared/s3"
+	"github.com/bwmarrin/snowflake"
+)
+
+var SnowflakeGen *snowflake.Node
+
+func init() {
+	// Snowflakes are used to identify jobs as they are created
+	// Snowflakes are unique IDs, similar to UUIDs, containing their creation
+	// timestamp and a few bits of metadata about what "node" was used to make
+	// the snowflake. Here, we use a random node number.
+	snow, err := snowflake.NewNode(rand.Int63n(1023))
+	if err != nil {
+		// throw up our hands
+		log.Fatal("Couldn't get a random int..." + err.Error())
+	}
+	SnowflakeGen = snow
+}
+
+// A batch of jobs created from an SQS message.
+type JobBatch struct {
+	// The minimal information for tracing the job back to an SQS message
+	Message JobMessage
+	// Jobs that were extracted from that message.
+	Jobs    []Job
+}
+
+// A job created from an SQS message and S3 record.
+type Job struct {
+	StartedAt           time.Time
+	Region, Bucket, Key string
+	SnowflakeId         string
+}
+
+// Minimal information to identify an SQS message
+type JobMessage struct {
+	ID            string
+	ReceiptHandle string
+	QueueUrl      string
+}
+
+// Turn an SQS message into a batch
+// TODO: should take a JobMessage
```

Comment:
I would say either implement the TODO items or just remove the comments for now.